compiler/rustc_borrowck/src/type_check/canonical.rs

```diff
@@ -165,6 +166,52 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
         result.unwrap_or(value)
     }
 
+    #[instrument(skip(self), level = "debug")]
+    pub(super) fn struct_tail(
+        &mut self,
+        ty: Ty<'tcx>,
+        location: impl NormalizeLocation,
+    ) -> Ty<'tcx> {
+        let tcx = self.tcx();
+        if self.infcx.next_trait_solver() {
+            let body = self.body;
+            let param_env = self.param_env;
+            self.fully_perform_op(
```

Comment:
This is incredibly gross, to the point that it kinda feels unnecessary? :thinking: But I really don't know how to simplify it, given the way that we need to wrap this in a type op and `struct_tail` doesn't have a fallible normalize fn...